### ISSUE TITLE
QT5 with dbus failing build process

### ISSUE THREAD

**Issue by manish2479** [spack/spack#5240]
Hi 

I am trying to build qt5 with dbus and it's failing with error.

```
qt/package.py:339, in configure:
     322      @when('@5.7:')
     323      def configure(self):
     324          config_args = self.common_config_args
     325
     326          if not sys.platform == 'darwin':
     327              config_args.extend([
     328                  '-qt-xcb',
     329              ])
     330
     331          if '~webkit' in self.spec:
     332              config_args.extend([
     333                  '-skip', 'webengine',
     334              ])
     335
     336          configure('-no-eglfs',
     337                    '-no-directfb',
     338                    '-{0}gtk'.format('' if '+gtk' in self.spec else 'no-'),
  >> 339                    *config_args)

```
Error message
```
spack/opt/spack/linux-sles12-x86_64/gcc-4.8/dbus-1.11.2-suvndtddjsenlhtqycnsympatohyhyer/lib -ldbus-1 -qt-xcb

This is the Qt Open Source Edition.

You are licensed to use this software under the terms of
the GNU Lesser General Public License (LGPL) version 3.
You are also licensed to use this software under the terms of
the GNU General Public License (GPL) version 2.

You have already accepted the terms of the Open Source license.

Creating qmake...
gmake: Nothing to be done for 'first'.

ERROR: Unknown command line option '-ldbus-1'.
```

Thank You,
                 Manish

**Comment by mathstuf** [comment]
It looks like those flags were added in 328b2142f5ea49ce07d6813073f37906b8a9c47d by @tgamblin to fix a Qt4 build failure. I suspect they should just be hidden behind a version check.

**Comment by bassenj** [comment]
In Qt5 `-ldbus-1` is configured in `qt-everywhere-src-5.10.0/qtbase/configure.json`. 
A version check solves the problem.
```python
        if '+dbus' in self.spec:
            dbus = self.spec['dbus'].prefix
            config_args.append('-dbus-linked')
            config_args.append('-I%s/dbus-1.0/include' % dbus.lib)
            config_args.append('-I%s/dbus-1.0' % dbus.include)
            config_args.append('-L%s' % dbus.lib)
            if '@4' in self.spec:
                config_args.append('-ldbus-1')
```

**Comment by alalazo** [comment]
This seems to be solved at 1d06949306b592c0f675dd959303ee1eda1a2cf2: 
```console
$ spack install qt+dbus
[ ... ]
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Successfully installed qt
  Fetch: 2.01s.  Build: 48m 1.11s.  Total: 48m 3.12s.
[+] /davide/home/userexternal/mculpo02/github/spack/opt/spack/linux-centos7-power8le/gcc-9.2.0/qt-5.13.1-q4arkyke2poojz5my2ssfywkghqzj7q7
```
